packages/plugin-http-client/src/rest/AbstractResource.ts

```diff
@@ -0,0 +1,170 @@
+import { GenericError, Dependencies } from '@ima/core';
+
+declare module '@ima/core' {
+  interface OCAliasMap {
+    REST_CLIENT_BASE_API_URL: string;
+  }
+}
+
+import {
+  HttpClient,
+  HttpClientRequestMethod,
+  HttpClientRequestOptions,
+} from '../HttpClient';
+
+export class AbstractResource {
+  #httpClient: HttpClient;
+  #baseApiUrl: string;
+
+  static get $dependencies(): Dependencies {
+    return [HttpClient, '?REST_CLIENT_BASE_API_URL'];
+  }
+
+  static get PathType() {
+    return {
+      GET: 'get',
+      CREATE: 'create',
+      UPDATE: 'update',
+      REPLACE: 'replace',
+      DELETE: 'delete',
+    };
+  }
+
+  get path(): { [pathType: string]: string } {
+    throw new GenericError(
+      `AbstractResource: Static getter "path" must be overriden.`
+    );
+  }
+
+  get entityClass() {
+    return null;
+  }
+
+  constructor(httpClient: HttpClient, baseApiUrl: string) {
+    this.#httpClient = httpClient;
+
+    if (!baseApiUrl) {
+      throw new GenericError(`REST_CLIENT_BASE_API_URL is not set.`);
+    }
+    this.#baseApiUrl = baseApiUrl;
+  }
+
+  update(
```

Comment:
Can we add `<B = any>` generics to all methods for proper body typing?

Reply:
Added.